File: 031.py
coins = [1, 2, 5, 10, 20, 50, 100, 200]
maxes = {1:200, 2:100, 5:40, 10:20, 20:10, 50:4, 100:2, 200:1}

count = 0

for a in range(201):
    for b in range(101):
        for c in range(41):
            for d in range(21):
                for e in range(11):
                    for f in range(5):
                        for g in range(3):
                            for h in range(2):
                                p = 1*a + 2*b + 5*c + 10*d + 20*e + 50*f + 100*g + 200*h
                                if p == 200:
                                    count += 1
    
print count

File: 031.c
#include <stdio.h>

int coins[8] = {1, 2, 5, 10, 20, 50, 100, 200};

int main(){

  int count = 0;

  for (int a=0; a < 2; a++) {
    for (int b=0; b < 3; b++) {
      for (int c=0; c < 5; c++) {
        for (int d=0; d < 11; d++) {
          for (int e=0; e < 21; e++) {
            for (int f=0; f < 41; f++) {
              for (int g=0; g < 101; g++) {
                for (int h=0; h < 201; h++) {
                  if (200*a + 100*b + 50*c + 20*d + 10*e + 5*f + 2*g + h == 200) {
                    count++;
                  }
                }
              }
            }
          }
        }
      }
    }
  }

  printf("combinations: %d\n", count);

  return 0;
}
